### PR TITLE
[Catalog] [Scaffolder] Analytics entity-awareness

### PR DESCRIPTION
**Pull request by iamEAP**
## What / Why

- Introduces a by-convention `entityRef` analytics context key (set on the `(Async)EntityProvider`) to allow events on catalog (and other) pages to be aggregated by entity.
- Adds a `create` event in the scaffolder whose subject is the name of the software created (falling backed to `new {templateName}` if for some reason the template doesn't ask for `name`). The template entity ref is set on the `entityRef` context.
- Also adds click handling on the Scaffolder stepper's "Next Step" buttons as a way to measure scaffolder form abandonment.

### Show me the data

The "create" event looks something like this:

Screenshot 2022-10-13 at 16 46 33

The step events look like this:

Screenshot 2022-10-13 at 16 48 19

And events underneath the analytics context resemble this:

Screenshot 2022-10-13 at 16 50 03

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by szubster** [comment]
Any chance to get this before next week release?

**Comment by iamEAP** [comment]
I too am keen to get this in ahead of the release!  Whaddya think @freben?

**Comment by iamEAP** [comment]
Or @Rugvip?

**Comment by freben** [comment]
I'm sorry I didn't manage to get to this sooner; there's a bit of a pile after the offsite.

**Comment by benjdlambert** [comment]
Looks fine to me - just a heads up though, that you might also want to implement this in the `/next` folder too :pray:

File: plugins/catalog-react/src/hooks/useEntity.tsx [hunk 1]
```diff
@@ -66,7 +67,13 @@ export const AsyncEntityProvider = ({
   // consumers might be doing things like `useContext(EntityContext)`
   return (
     <NewEntityContext.Provider value={createVersionedValueMap({ 1: value })}>
-      {children}
+      <AnalyticsContext
+        attributes={{
+          ...(entity ? { entityRef: stringifyEntityRef(entity) } : undefined),
```
Comment (freben) [inline plugins/catalog-react/src/hooks/useEntity.tsx:72]: just a note that this attributes object isn't memo'd, but that probably doesn't matter much here

File: plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx [hunk 2]
```diff
@@ -123,6 +126,8 @@ export const MultistepJsonForm = (props: Props) => {
     finishButtonLabel,
     layouts,
   } = props;
+  const { templateName } = useRouteRefParams(selectedTemplateRouteRef);
```
Comment (freben) [inline plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx:129]: I see it's just used for a fallback label text so it's not a big deal, but just wanted to note here that it doesn't include the namespace.

**Review by freben — APPROVED** [review]
Looks great. Just two comments that can safely be ignored.

**Comment by iamEAP** [comment]
Will cover `next` in a follow up, @benjdlambert

**Pull request merged**